.github/workflows/ci.yml

```diff
@@ -8,72 +8,50 @@ on:
   pull_request:
   workflow_dispatch:
 
-env:
-  POETRY_VERSION: 1.8.4
-
 jobs:
   flake8:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
-      - name: Setup Python
+      - name: Setup Python 3.13
         uses: actions/setup-python@v5
         with:
-          python-version: '3.12'
+          python-version: '3.13'
 
-      - name: Install poetry
-        uses: snok/install-poetry@v1
-        with:
-          version: ${{ env.POETRY_VERSION }}
-          virtualenvs-create: false
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v4
 
-      # Skip cache because we have few deps
-      - name: Install flake8
-        run: poetry install -v --no-root --only lint
+      - name: Install project + dev deps
+        run: uv sync --extra dev
```

Comment:
Please commit the uv lock file for reproducible environments (locally and in CI).